test/MyFirstApiConsume.test.js

```diff
@@ -0,0 +1,98 @@
+const axios = require('axios');
+const { expect } = require('chai');
+const { StatusCodes } = require('http-status-codes');
+
+axios.interceptors.request.use((request) => {
+	const headers = {
+		...request.headers.common,
+		...request.headers[request.method],
+		...request.headers,
+	};
+
+	// remove unrelevant headers to reduce the noise
+	['common', 'get', 'post', 'head', 'put', 'patch', 'delete'].forEach(
+		(header) => {
+			delete headers[header];
+		}
+	);
+
+	const printable = `
+---------------------------------------------- \n
+REQUEST ⏩ \n
+${new Date()}
+Request: ${request.method.toUpperCase()}
+URL:${request.url}
+Headers: ${JSON.stringify(headers, null, 2)} \n
+Data: ${
+		request.data ? JSON.stringify(response.data, null, 2) : 'There is no data'
+	}
+	`;
+
+	console.log(printable);
+	return request;
+});
+
+axios.interceptors.response.use((response) => {
+	const printable = `
+⏩ RESPONSE \n
+${new Date()}
+Status: ${response.status} ${response.statusText}
+Headers: ${JSON.stringify(response.headers, null, 2)} \n
+Data: ${
+		response.data ? JSON.stringify(response.data, null, 2) : 'There is no data'
+	}
+`;
+
+	console.log(printable);
+	return response;
+});
+
+describe('First Api Tests', () => {
+	it('Consume GET Service', async () => {
+		const response = await axios.get('https://httpbin.org/ip');
+
+		expect(response.status).to.equal(StatusCodes.OK);
+		expect(response.data).to.have.property('origin');
+	});
+
+	it('Consume GET Service with query parameters', async () => {
+		const query = {
+			name: 'John',
+			age: '31',
+			city: 'New York',
+		};
+
+		const response = await axios.get('https://httpbin.org/get', { query });
+
+		expect(response.status).to.equal(StatusCodes.OK);
+		expect(response.config.query).to.eql(query);
+	});
+
+	it('Consume HEAD Service', async () => {
+		const response = await axios.get('https://httpbin.org/headers');
+
+		expect(response.status).to.equal(StatusCodes.OK);
+		expect(response.data).to.have.property('headers');
```

Comment:
Remember to send as head method
```suggestion
		const response = await axios.head('https://httpbin.org/headers');

		expect(response.status).to.equal(StatusCodes.OK);
		expect(response.data).to.have.property('headers');
```